ml-agents/mlagents/trainers/agent_processor.py

```diff
@@ -112,21 +136,65 @@ def add_experiences(
                         [_gid], take_action_outputs["action"]
                     )
 
+    def _gather_group_obs(
+        self, step: Union[TerminalStep, DecisionStep], worker_id: int
+    ) -> None:
+        global_agent_id = get_global_agent_id(worker_id, step.agent_id)
+        stored_decision_step, idx = self.last_step_result.get(
+            global_agent_id, (None, None)
+        )
+        stored_take_action_outputs = self.last_take_action_outputs.get(
+            global_agent_id, None
+        )
+        if stored_decision_step is not None and stored_take_action_outputs is not None:
+            if step.group_id > 0:
```

Comment:
This is using implicit assumption that 0 means no group. I'd add some comments or put it into a separate function

Reply:
Added a comment - we can add a utility method too, `step.has_agent_group`?

Reply:
yeah I was thinking more like a utility method

Reply:
Ah yeah that's not a bad idea. Let's do a separate PR to add this, and add the group ID to the documentation for the Python API